### PR TITLE
Assign proper gpio pins to nRF52-DK

### LINKED ISSUE #1079: nRF52-DK broken on `master`
Hey,

`nRF52-DK` seems to be broken on the latest master (f695769e34b2b67073c7f02518b166c537111359)

I can see the following print: 
```bash
Initialization complete. Entering main loop
NRF52-DK HW INFO: Variant: AAB0, Part: N52832, Package: QF, Ram: K64, Flash: K512
```

But then `LED3` and `LED4` lights up even without any `app` loaded!

Anyone that can produce this?

### PR DESCRIPTION
**Pull request by niklasad1**
### Pull Request Overview

In `nrf52dk_base` we had assumed some pins are the same but they were
not. Because for `nRF52DK` pin 19 and pin 20 were assigned to both the
LEDS and the SPI drivers!

However, now `setup_board` takes very many parameters which is not
efficient i.e, can't go in the registers but since this is during
the initialization I guess it is ok!

Closes #1079 

### Testing Strategy

This pull request was tested on `nRF52-DK`

### TODO or Help Wanted

- [x] test on nRF52840DK

### Documentation Updated

~- [ ] Updated the relevant files in `/docs`, or no updates are required.~

### Formatting

- [x] Ran `make formatall`.

### CODE REVIEW

**Comment by niklasad1** [comment]
> What do you mean "not efficient" and "can't go into the registers"? Do you mean the arguments passed to the common intializer don't fit in registers and therefore it's probably wasting stack space? My guess is actually that it's likely just inlined, so doesn't make a difference in practice.

Yupp, I meant there too many arguments to go in the registers and goes via the stack instead! But, bad phrasing from my side I should have been more explicit. If I'm not mistaken I think ARM passes the first four arguments into r0-r3 (assuming 32-bit sized) and the rest on the stack. But, you are probably right w.r.t inlining (haven't checked the assembly)

> It is getting a bit messy, and might be worth having a utlity struct or something to make it more readable, but i think it's fine for this pull request.

Agree!

**Comment by phil-levis** [comment]
> On Jul 8, 2018, at 10:49 AM, Niklas Adolfsson <notifications@github.com> wrote:
> 
> But, you are probably right w.r.t inlining (haven't checked the assembly)

I’d caution from assuming the compiler will make optimizations for you. It’s good when it does, but if there’s a big cost when it doesn’t then you want to preclude it. E.g., make sure arguments don’t spill to the stack by only having 4. 

Otherwise you start to get weird failure cases. For example, a change in one part of the kernel happens to change inlining in some somewhat related code, which then causes stack overflows. The exact same call chain succeeds in one kernel but fails in the other, with no code changes in any of the functions in the call chain.

Phil

File: boards/nordic/nrf52dk_base/src/lib.rs [hunk 1]
```diff
@@ -14,6 +14,53 @@ use capsules::virtual_spi::MuxSpiMaster;
...
+    device: usize,
+    client: usize,
+    client_sector: usize,
+}
+
+impl SpiMX25R6435FPins {
+    pub fn new(device: usize, client: usize, client_sector: usize) -> Self {
+        Self {
+            device,
+            client,
```
Comment (bradjc) [inline boards/nordic/nrf52dk_base/src/lib.rs]: This should be `write_protect_pin`.

File: boards/nordic/nrf52dk_base/src/lib.rs [hunk 2]
```diff
@@ -14,6 +14,53 @@ use capsules::virtual_spi::MuxSpiMaster;
...
+            client_sector,
```
Comment (bradjc) [inline boards/nordic/nrf52dk_base/src/lib.rs]: This should be `hold_pin`.

**Comment by alevy** [comment]
@bradjc renamed.

File: boards/nordic/nrf52840dk/src/main.rs [hunk 3]
```diff
@@ -130,10 +145,16 @@ pub unsafe fn reset_handler() {
         LED2_PIN,
         LED3_PIN,
         led_pins,
+        &UartPins::new(UART_RTS, UART_TXD, UART_RXD, UART_CTS),
+        &SpiPins::new(SPI_MOSI, SPI_MISO, SPI_CLK),
+        &Some(SpiMX25R6435FPins::new(
+            SPI_MX25R6435F_DEVICE,
+            SPI_MX25R6435F_CLIENT,
```
Comment (bradjc) [inline boards/nordic/nrf52840dk/src/main.rs]: I was hoping you would fix these as well :)

Comment (alevy) [reply boards/nordic/nrf52840dk/src/main.rs]: HAPPY NOW???!!!!!

**Pull request merged**